src/shared/codelens/pythonCodeLensProvider.ts

```diff
@@ -51,12 +52,40 @@ const getSamProjectDirPathForFile = async (filepath: string): Promise<string> =>
     return path.dirname(filepath)
 }
 
-const getLambdaHandlerCandidates = async ({ uri }: { uri: vscode.Uri }): Promise<LambdaHandlerCandidate[]> => {
+async function getLambdaHandlerCandidates({
+    uri,
+    pythonSettings
+}: {
+    uri: vscode.Uri
+    pythonSettings: SettingsConfiguration
+}): Promise<LambdaHandlerCandidate[]> {
+    const PYTHON_JEDI_ENABLED_KEY = 'jediEnabled'
+    const RETRY_INTERVAL_MS = 1000
+    const MAX_RETRIES = 10
+
     const logger = getLogger()
     const filename = uri.fsPath
-    const symbols: vscode.DocumentSymbol[] =
-        (await vscode.commands.executeCommand<vscode.DocumentSymbol[]>('vscode.executeDocumentSymbolProvider', uri)) ||
-        []
+
+    let symbols: vscode.DocumentSymbol[] =
+        await vscode.commands.executeCommand<vscode.DocumentSymbol[]>('vscode.executeDocumentSymbolProvider', uri) || []
+
+    // A recent regression in vscode-python stops codelenses from rendering if we first return an empty array
```

Comment:
VS Code or Python specifically?

Reply:
We can't say for sure, since we can't use the latest version of the Python extension on older versions of VS Code. But the fact that the issue only affects Python suggests that it is the Python extension.

Once I file a bug upstream, I will add a link.